### ISSUE TITLE
Can p5 be used as headless library?

### ISSUE THREAD

**Issue by lanyusan** [p5py/p5#339]
I am using pycairo as headless canvas. But its API isn't compatible with processing's.

If P5 can be used as headless canvas, it would be great.

If yes, how to use it? I couldn't find such information from doc.

Thanks

**Comment by tushar5526** [comment]
There isn't direct support as of now for p5py to be used headlessly, you could however use virtual frame buffers.

**Comment by martynvdijke** [comment]
How would one go about this ?
I basically want to generate (for now) an image headless grab the image and save to png for usage.

**Comment by martynvdijke** [comment]
I did some digging, I can some a long way by running the rendering process in a separate thread.
But I cannot find a way to prevent the preview window to pop up

**Comment by ivanlen** [comment]
Is there any progress on this front?
I would like to use p5 in a jupyter notebook for example and render the images there somehow, is that possible?

**Comment by tushar5526** [comment]
@ivanlen it's still a WIP, but you can checkout https://py5coding.org/index.html which supports this.

**Comment by ivanlen** [comment]
> @ivanlen it's still a WIP, but you can checkout https://py5coding.org/index.html which supports this.
@tushar5526 

Yes thanks!, already saw that one but they are using ipython kernel as backend which is not currently compatible with ARM mac architecture. I was able to make work `p5py` with mac and thats why I interested in using it that way.

**Comment by tushar5526** [comment]
@ivanlen one of our 2D rendering backend skia supports this feature - but it's still a WIP and I am not sure if I can find time for it soon.
